scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Provides query functionality for .csv metrics from S3 buckets
+"""
+
+import argparse
+import csv
+import os
+
+from boto3.session import Session
+
+BUCKET_NAME = "o3de-metrics"
+
+SESSION = None
+DOWNLOADED_FILES = []
+
+
+def _get_session():
+    """
+    Gets an AWS session via boto3 or returns the existing one
+    """
+    global SESSION
+    if SESSION is None:
+        # get session
+        id_ = os.environ['AWS_ACCESS_KEY_ID']
+        secret = os.environ['AWS_SECRET_ACCESS_KEY']
+        token = os.environ['AWS_SESSION_TOKEN']
+
+        SESSION = Session(
+            aws_access_key_id=id_,
+            aws_secret_access_key=secret,
+            aws_session_token=token,
+            region_name='us-west-2'
+        )
+    return SESSION
+
+
+def _download_csvs(branch, start_week, end_week, download_folder):
+    """
+    Downloads the desired csvs based on branch, start_week, and end_week to the download_folder
+    :param branch: name of the desired branch
+    :param start_week: number of the first week to include
+    :param end_week: number of the last week to include
+    :param download_folder: destination folder for the .csvs to be stored while they are being parsed
+    """
+    bucket = _get_session().resource('s3').Bucket(BUCKET_NAME)
+
+    prefix = "csv/" + branch
+
+    objs = list(bucket.objects.filter(Prefix=prefix))
+
+    file_type = ".csv"
+    for obj in objs:
+        if file_type == obj.key[-4:]:
+            try:
+                # Get the key file values
+                filter_values = obj.key.split('/')
+                branch = filter_values[-3]
```

Comment:
Also I think the "branch" variable is being overshadowed by the function argument